tools/sof-dump-status.py

```diff
@@ -437,6 +437,35 @@ def dump_dapm(dapm, filter = "all"):
             print(run_status['status'])
         exit(0)
 
+    if ret_args['export'] is not None:
+        sysinfo.loadProcSound()
+        pipeline_lst = []
+        for (card_id, card_info) in sysinfo.proc_card.items():
+            for pcm in card_info['pcm']:
+                # There are limited pipeline parameters in the proc file system,
+                # add some default parameters to make use of sof-test for legacy HDA test
+                pcm['fmt'] = 'S16_LE'
+                pcm['fmts'] = 'S16_LE S24_LE S32_LE'
+                pcm['rate'] = '48000'
+                pcm['channel'] = '2'
+                pcm['dev'] = 'hw:{},{}'.format(card_id, pcm['id'])
+            pipeline_lst.extend(card_info['pcm'])
+        # The filter string may be very complex due to the compatibility with topology pipeline
+        # filter string, but we only implement a simple type filter here.
+        try:
+            # extract requested pipeline type from filter string, and ignore others.
+            # eg, the requested_type from filter string 'type:capture & pga:any | id:3' is 'capture'
+            requested_type = [f.split(':')[1] for f in ret_args['export'].split(' ') if f.strip().startswith('type')][0]
```

Comment:
I still think one intermediate variable would make this much easier to read but the comment makes it very clear, thank you.

```python
filter_elements = ret_args['export'].split(' ')
first_requested_type = [f.split(':')[1] for f in filter_elements
    if f.strip().startswith('type')][0]
```